src/matter/cluster/OperationalCredentialsCluster.ts

```diff
@@ -4,93 +4,162 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { Cluster, Command, NoResponseT } from "./Cluster";
+import { Attribute, Cluster, Command, NoResponseT } from "./Cluster";
 import { TlvType } from "../../codec/TlvCodec";
-import { ByteStringT, Field, ObjectT, OptionalField, Template, UnsignedIntT, UnsignedLongT } from "../../codec/TlvObjectCodec";
+import { ByteStringT, Field, ObjectT, OptionalField, Template, UnsignedIntT, UnsignedLongT, StringT, ArrayT, BooleanT } from "../../codec/TlvObjectCodec";
+
+const FabricDescriptorT = ObjectT({
+    rootPublicKey: Field(1, StringT), /* length: 65 */
+    vendorId: Field(2, UnsignedIntT), /* type: vendor-id */
+    fabricID: Field(3, UnsignedIntT), /* type: fabric-id */
+    nodeID: Field(4, UnsignedIntT), /* type: node-id */
+    label: Field(5, StringT), /* maxLength: 32, default: "" */
+});
+
+const NocT = ObjectT({
+    noc: Field(1, StringT), /* maxLength: 400 */
+    icac: Field(2, StringT), /* maxLength: 400, default(not present): null */
+});
 
 export const enum CertificateType {
     DeviceAttestation = 1,
     ProductAttestationIntermediate = 2,
 }
 export const CertificateTypeT = { tlvType: TlvType.UnsignedInt } as Template<CertificateType>;
 
-export const RequestWithNonceT = ObjectT({
-    nonce: Field(0, ByteStringT),
+export const AttestationRequestT = ObjectT({
+    attestationNonce: Field(0, ByteStringT), /* length: 32 */
```

Comment:
Another question: keep the propoerty names near to the specs or simplyify and use the "cluster or message type" as reference and so simplify - as you did with just calling it "nonce"? has all it's pro and cons ... I personally think staying near to specs makes it easier to map code later to specs

Reply:
The issue with such Spec references is that they might change (if they ad d a section with new number then we run out of sync very fast. In fact the only reference that makes sense is the obvious one "see "Operational Credentials Cluster" specification in Matter Core Specs (in fcat withthe 3 PDFs I'm sonstantly searching and switching because all stuff is in different places ;-))) But in fact this is the obvious reference. ... so no idea if refernce make sense if we keep the naming because then you can also search for type names or such in the PDFs and have the place